app/Nova/Actions/Payments/RecordPayment.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Nova\Actions\Payments;
+
+use App\Models\DuesTransaction;
+use App\Models\Payable;
+use App\Models\Payment;
+use App\Models\TravelAssignment;
+use Exception;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+use Laravel\Nova\Actions\Action;
+use Laravel\Nova\Fields\ActionFields;
+use Laravel\Nova\Http\Requests\NovaRequest;
+
+abstract class RecordPayment extends Action
+{
+    /**
+     * Indicates if this action is only available on the resource detail view.
+     *
+     * @var bool
+     */
+    public $onlyOnDetail = true;
+
+    /**
+     * Indicates if need to skip log action events for models.
+     *
+     * @var bool
+     */
+    public $withoutActionEvents = true;
```

Comment:
It disables creating action logs. Since this action either fails or creates an object already, it doesn't seem useful.

Actions don't show on payables anyway, I'm not sure if that disables action logs as well.